src/ethstat.c

```diff
@@ -41,38 +42,108 @@
 #if HAVE_LINUX_ETHTOOL_H
 #include <linux/ethtool.h>
 #endif
+#if HAVE_IFADDRS_H
+#include <ifaddrs.h>
+#endif
 
 struct value_map_s {
   char type[DATA_MAX_NAME_LEN];
   char type_instance[DATA_MAX_NAME_LEN];
 };
 typedef struct value_map_s value_map_t;
 
-static char **interfaces;
-static size_t interfaces_num;
+static char **regexed_interfaces = NULL;
+static size_t regexed_interfaces_num = 0;
+
+static char **valid_interfaces = NULL;
+static size_t valid_interfaces_num = 0;
 
 static c_avl_tree_t *value_map;
 
 static bool collect_mapped_only;
 
+static int ethstat_check_if_existing_interface(char **existing_interfaces,
+                                               char *entry) {
+  char *aux;
+  int idx;
+
+  for (idx = 0; idx < valid_interfaces_num; ++idx) {
+    aux = strdup(existing_interfaces[idx]);
+    if (strcmp(aux, entry) == 0)
```

Comment:
It seems you like memory leaks much. I don't see why you need to do `strdup` at all.